src/components/include/utils/timer_thread.h

```diff
@@ -393,10 +393,13 @@ void TimerThread<T>::TimerDelegate::threadMain() {
 template <class T>
 void TimerThread<T>::TimerLooperDelegate::threadMain() {
   using sync_primitives::ConditionalVariable;
-  sync_primitives::AutoLock auto_lock(TimerDelegate::state_lock_);
+  sync_primitives::Lock lock;
+  lock.Acquire();
```

Comment:
What the reason to create local ``lock`?
It will be the different for each call.